### ISSUE TITLE
Single issue added repeatedly with multiple github syncs

### ISSUE THREAD

**Issue by djmitche** [ralphbean/bugwarrior#220]
Config snippet:

```
[github_relengapi]
service = github
github.username = mozilla
github.include_repos = build-relengapi, build-tooltool
only_if_assigned = djmitche
also_unassigned = False
github.login = djmitche
github.password = ...
github.import_labels_as_tags = True
github.label_template = github_{{label}}
default_priority = M 
add_tags = relengapi
description_template = ({{githubrepo}} #{{githubnumber}}) {{githubtitle}}
project_template = moz 

[github_fwunit]
service = github
github.username = mozilla
github.include_repos = build-fwunit
only_if_assigned = djmitche
github.login = djmitche
github.password = ...
github.import_labels_as_tags = True
github.label_template = github_{{label}}
default_priority = M 
add_tags = fwunit
description_template = ({{githubrepo}} #{{githubnumber}}) {{githubtitle}}
project_template = moz 

[github_bits]
service = github
github.username = djmitche
github.include_repos = webmock, prettyip, pynsca
only_if_assigned = djmitche
github.login = djmitche
github.password = ...
github.import_labels_as_tags = True
github.label_template = github_{{label}}
default_priority = M 
add_tags = bits
description_template = ({{githubrepo}} #{{githubnumber}}) {{githubtitle}}
project_template = 
```

is producing:

``````
Name             Value                                                              
ID               -
Description      (djmitche/prettyip #6) Except is sometimes too broad               
                   2015-06-05 13:50:19 https://github.com/djmitche/prettyip/issues/6
Status           Completed
Project          moz                                                                
Entered          2015-06-05 13:50:19 (3 hours)
End              2015-06-05 14:50:18                                                
Last modified    2015-06-05 14:50:18 (2 hours)
Tags             relengapi                                                          
Virtual tags     ANNOTATED COMPLETED TAGGED UNBLOCKED
UUID             51376e4a-5e9d-4b46-8d8b-222532ea63b9                               
Urgency           4.2
Github Body      expected:                                                          
                 ```                                                                
                 10.132.30.0/24 except 10.132.30.90, 10.132.30.96                   
                 ```                                                                
                 got:                                                               
                 ```                                                                
                 10.132.30.0/23 except 10.132.30.90, 10.132.30.96, 10.132.31.0/24   
                 ```                                                                
                 (which left me scratching my head for a while!)                    
Github Created   2015-06-05 10:52:17
Github Issue/PR  6                                                                  
Github Repo Slug djmitche/prettyip
Github Title     Except is sometimes too broad                                      
Github Type      issue
Github Updated   2015-06-05 10:52:20                                                
Github URL       https://github.com/djmitche/prettyip/issues/6
Priority         M                                                                  

    project             1 *    1 =      1
    tags              0.8 *    1 =    0.8
    UDA priority.M      1 *  3.9 =    3.9
    TAG relengapi       1 * -1.5 =   -1.5
                                   ------
                                      4.2

Date                Modification                                                        
2015-06-05 13:50:19 Annotation of 'https://github.com/djmitche/prettyip/issues/6' added.
2015-06-05 14:50:18 End set to '2015-06-05 14:50:18'.                                   
                    Status changed from 'pending' to 'completed'.                       


Name             Value                                                              
ID               -
Description      (djmitche/prettyip #6) Except is sometimes too broad               
                   2015-06-05 13:50:20 https://github.com/djmitche/prettyip/issues/6
Status           Completed
Project          moz                                                                
Entered          2015-06-05 13:50:20 (3 hours)
End              2015-06-05 14:50:18                                                
Last modified    2015-06-05 14:50:18 (2 hours)
Tags             fwunit                                                             
Virtual tags     ANNOTATED COMPLETED TAGGED UNBLOCKED
UUID             e707cfc4-cd4e-4649-8ce7-4173aad214e6                               
Urgency           5.7
Github Body      expected:                                                          
                 ```                                                                
                 10.132.30.0/24 except 10.132.30.90, 10.132.30.96                   
                 ```                                                                
                 got:                                                               
                 ```                                                                
                 10.132.30.0/23 except 10.132.30.90, 10.132.30.96, 10.132.31.0/24   
                 ```                                                                
                 (which left me scratching my head for a while!)                    
Github Created   2015-06-05 10:52:17
Github Issue/PR  6                                                                  
Github Repo Slug djmitche/prettyip
Github Title     Except is sometimes too broad                                      
Github Type      issue
Github Updated   2015-06-05 10:52:20                                                
Github URL       https://github.com/djmitche/prettyip/issues/6
Priority         M                                                                  

    project             1 *    1 =      1
    tags              0.8 *    1 =    0.8
    UDA priority.M      1 *  3.9 =    3.9
                                   ------
                                      5.7

Date                Modification                                                        
2015-06-05 13:50:20 Annotation of 'https://github.com/djmitche/prettyip/issues/6' added.
2015-06-05 14:50:18 End set to '2015-06-05 14:50:18'.                                   
                    Status changed from 'pending' to 'completed'.                       


Name             Value                                                              
ID               -
Description      (djmitche/prettyip #6) Except is sometimes too broad               
                   2015-06-05 13:50:19 https://github.com/djmitche/prettyip/issues/6
Status           Completed
Entered          2015-06-05 13:50:19 (3 hours)                                      
End              2015-06-05 14:50:18
Last modified    2015-06-05 14:50:18 (2 hours)                                      
Tags             bits
Virtual tags     ANNOTATED COMPLETED TAGGED UNBLOCKED                               
UUID             d8ed298f-871e-4560-9e10-0af49b79b805
Urgency           4.7                                                               
Github Body      expected:
                 ```
                 10.132.30.0/24 except 10.132.30.90, 10.132.30.96
                 ```
                 got:
                 ```
                 10.132.30.0/23 except 10.132.30.90, 10.132.30.96, 10.132.31.0/24
                 ```
                 (which left me scratching my head for a while!)
Github Created   2015-06-05 10:52:17                                                
Github Issue/PR  6
Github Repo Slug djmitche/prettyip                                                  
Github Title     Except is sometimes too broad
Github Type      issue                                                              
Github Updated   2015-06-05 10:52:20
Github URL       https://github.com/djmitche/prettyip/issues/6                      
Priority         M

    tags              0.8 *    1 =    0.8
    UDA priority.M      1 *  3.9 =    3.9
                                   ------
                                      4.7

Date                Modification                                                        
2015-06-05 13:50:19 Annotation of 'https://github.com/djmitche/prettyip/issues/6' added.
2015-06-05 14:50:18 End set to '2015-06-05 14:50:18'.                                   
                    Status changed from 'pending' to 'completed'.                       
``````

and subsequently failing with "ERROR:db:Multiple matches".

Judging by the tags in those tasks, it looks like all three config blocks are matching this task, yet only one should (github_bits).

**Comment by djmitche** [comment]
I haven't seen this repeated with any other github issue.. maybe just a weird one?  Or some kind of hash collision?

**Comment by djmitche** [comment]
Today I saw it with a [new issue](https://github.com/taskcluster/relengapi-proxy/pull/7):

```
ERROR:db:Multiple matches: Issue (taskcluster/relengapi-proxy #7) Some README fixes matched multiple IDs: set([UUID('24492d13-4f49-40d8-bdac-3909f3912523'), UUID('f64dce46-ff00-4620-a9f7-4be9aa038e5a'), UUID('f8ff2600-8e4d-42db-978e-c6fb327a825a')])
ERROR:db:Multiple matches: Issue (taskcluster/relengapi-proxy #7) Some README fixes matched multiple IDs: set([UUID('24492d13-4f49-40d8-bdac-3909f3912523'), UUID('f64dce46-ff00-4620-a9f7-4be9aa038e5a'), UUID('f8ff2600-8e4d-42db-978e-c6fb327a825a')])
ERROR:db:Multiple matches: Issue (taskcluster/relengapi-proxy #7) Some README fixes matched multiple IDs: set([UUID('24492d13-4f49-40d8-bdac-3909f3912523'), UUID('f64dce46-ff00-4620-a9f7-4be9aa038e5a'), UUID('f8ff2600-8e4d-42db-978e-c6fb327a825a')])
```

Any idea what it is about these issues that causes the repetition?  I have lots of other issues and PRs in taskwarrior without any trouble.

**Comment by ralphbean** [comment]
We may have already tried this before.. but do you have the following in the ``[general]`` section of your config?

```
legacy_matching = False
```

**Comment by djmitche** [comment]
I do, yes, on your advice around the end of September :(

**Comment by ralphbean** [comment]
Crumb.  This looks weird indeed!  If you manually all three uuids, does it get back into the same state with subsequent bugwarrior runs?

**Comment by djmitche** [comment]
I assume you mean "manually delete" -- I tried that, but unfortunately I merged the PR at the same time, so no good data there.  I'll try next time.

**Comment by ralphbean** [comment]
Ha, yes I did.  Typos, typos.

**Comment by djmitche** [comment]
I have a PR now which is appearing multiple times because I have multiple github syncs set up.  I suspect that's at the root of the issue.

```
ID  Proj Pri Age   Urg  Description                                                                                           Tags     
166 moz  H   44min  7.8 (taskcluster/taskcluster-lib-api #8) [WIP] support temp creds with clientId != ext.certificate.issuer fwunit   
162      H   44min  6.8 (taskcluster/taskcluster-lib-api #8) [WIP] support temp creds with clientId != ext.certificate.issuer bits     
164 moz  H   44min  6.3 (taskcluster/taskcluster-lib-api #8) [WIP] support temp creds with clientId != ext.certificate.issuer relengapi

```
```
[github_relengapi]
service = github
github.username = mozilla
github.include_repos = build-relengapi, build-tooltool
only_if_assigned = djmitche
# don't include unassigned stuff - there's just too much!
also_unassigned = False
github.login = djmitche
github.import_labels_as_tags = True
github.label_template = github_{{label}}
default_priority = M
add_tags = relengapi
description_template = ({{githubrepo}} #{{githubnumber}}) {{githubtitle}}
project_template = moz

[github_fwunit]
service = github
github.username = mozilla
github.include_repos = build-fwunit
only_if_assigned = djmitche
#also_unassigned = True
github.login = djmitche
github.import_labels_as_tags = True
github.label_template = github_{{label}}
default_priority = M
add_tags = fwunit
description_template = ({{githubrepo}} #{{githubnumber}}) {{githubtitle}}
project_template = moz

[github_bits]
service = github
github.username = djmitche
github.include_repos = webmock, prettyip, pynsca, proj, exchanger
only_if_assigned = djmitche
github.login = djmitche
github.import_labels_as_tags = True
github.label_template = github_{{label}}
default_priority = M
add_tags = bits
description_template = ({{githubrepo}} #{{githubnumber}}) {{githubtitle}}
project_template = 
```

**Comment by ssoriche** [comment]
I've been having this issue recently too. I've discovered that the `project_template` attribute appears to be applied to every stanza that's using GitHub. Either I end up with database errors for duplicate entries, or tasks are switching between projects. As soon as I remove the `project_template` configuration, the tasks are assigned to the correct repo, and there are no errors.

**Comment by ryneeverett** [comment]
Triage: I'm guessing this is resolved, perhaps by #654
